src/routing.js

```diff
@@ -44,21 +44,32 @@ export const queryStringMiddleware = (history, {reduxPathname, routes}, config =
         const result = next(action);
         const state = store.getState();
         const currentPath = reduxPathname(state);
-        const routeConfig = routes[currentPath];
+        const routesConfig = !_.isArray(routes[currentPath]) ? [routes[currentPath]] : routes[currentPath];
 
-        if (routeConfig) {
-            const {listen, select, serialize} = routeConfig;
-            const qsState = select(state);
+        if (!_.isEmpty(routesConfig)) {
+            let queryString = [];
+            routesConfig.forEach((routeConfig) => {
+                if (routeConfig) {
+                    const {select, serialize} = routeConfig;
+                    const qsState = select(state);
 
-            if (qsState === null) {
-                return result;
-            }
-            let data = _pruneNullLeaves(qsState);
-            if (listen === action.type) {
-                if (serialize) {
-                    data = _applySerialization(data, serialize);
+                    if (qsState === null) {
+                        return result;
+                    }
+                    let data = _pruneNullLeaves(qsState);
+                    if (_.contains(routesConfig.map((route) => (route.listen)), action.type)) {
+                        if (serialize) {
+                            data = _applySerialization(data, serialize);
+                        }
+                        if (!_.isEmpty(data)) {
+                            queryString.push(qs.stringify(data, SERIALIZATION_OPTIONS));
+                        }
+                    }
                 }
-                const queryString = qs.stringify(data, SERIALIZATION_OPTIONS);
+                return result;
+            });
+            queryString = queryString.join('&');
```

Comment:
Why not accumulating everything in an object and only do the "stringifying" at the end? It's true that the attributes would be overridden, but a) having >1 listener per action would be useless and b) what would be the point of two listeners setting the same query argument?